### DISCUSSION TITLE
Spotify reg issue with IPV6 device

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#1061] [Fixed Issues]
Running rAudio 20221010

After registering Spotify, I noticed the following error in the spotifyd info window (I could not see rAudio as a device from either a PC or iPhone via Spotify app).

[spotifyd reg issue.txt](https://github.com/rern/rAudio-1/files/9756459/spotifyd.reg.issue.txt)

Viewing the system status log, it is clear that only IPV4 is still used and that IPV6 has been disabled.

Therefore, can spotifyd be forced to use IPV4 or am I missing something in the configuration steps?

Also, must the username / password be entered into the spotifyd.conf file or does the registration process (with secret key) handle the authentication?

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by HunLucas** [comment]
following, same issue

**Comment by R101G** [comment]
I think there was a reason for IPv6 being disabled in /boot/cmdline.txt, but you can enable it if you need it. I enabled it out of interest, and did not see any disadvantage.

**Reply by HunLucas** [reply]
Thanks, the error message about ipv6 is gone, but still can't see the rAudio in the spotify's device list.

spotifyd.service- A spotify playing daemon
     Loaded: loaded (/etc/systemd/system/spotifyd.service; enabled; preset: disabled)
     Active: active (running) since Sun 2023-01-29 11:27:29 UTC; 4s ago
   Main PID: 2123 (spotifyd)
      Tasks: 2 (limit: 4524)
        CPU: 27ms
     CGroup: /system.slice/spotifyd.service
             └─2123 /usr/bin/spotifyd --no-daemon

Jan 29 11:27:29 rAudio systemd[1]: Started A spotify playing daemon.
Jan 29 11:27:29 rAudio spotifyd[2123]: Loading config from "/etc/spotifyd.conf"
Jan 29 11:27:29 rAudio spotifyd[2123]: No username specified. Checking username_cmd
Jan 29 11:27:29 rAudio spotifyd[2123]: No username_cmd specified
Jan 29 11:27:29 rAudio spotifyd[2123]: No password specified. Checking password_cmd
Jan 29 11:27:29 rAudio spotifyd[2123]: No password_cmd specified
Jan 29 11:27:29 rAudio spotifyd[2123]: No proxy specified
Jan 29 11:27:29 rAudio spotifyd[2123]: Using software volume controller.
Jan 29 11:27:29 rAudio spotifyd[2123]: no usable credentials found, enabling discovery

**Comment by McPaHa** [comment]
Same problem - cannot use spotify... following

**Comment by foberleitner** [comment]
Also following because having the same issue: can not use Spotify.
Enabling IPv6 in `/boot/cmdline.txt` did not resolve the problem.

Also noticed that Airplay stopped working - I assume that the root cause is a mechanism/service shared by both Spotify and Airplay, maybe how these are discovered on LAN.
But that is just a guestimate as I have no knowledge about that domain (networking etc.)

**Comment by Sagnasty** [comment]
I rebuilt all my systems directly from image 20231022 and did NOT restore any backups from previous rAudio versions (was previously running 20230620).

I can now report that I have Spotify registered / working  on all my pi's using IPv4 addresses (no errors referencing IPv6 during the registration process).

Nothing has changed other than a fresh install of the 20231022 image.

Playback with Spotify appears to be working fine with iPhone running iOS 17.0.3 on my local IPv4 network.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+
